### DISCUSSION TITLE
Only load reCAPTCHA if cookie consent was granted

### DISCUSSION THREAD

**Discussion by thisisjamessmith** [verbb/formie#2061] [Q&A]
Hiya, 

Is there any way that we can delay the injection/execution of reCAPTCHA scripts unless/until consent has been granted?

**Comment by engram-design** [comment]
Not at the moment, sorry. Formie 3 introduces a way to only load scripts and initialize captchas when visible on the page, so we can work from that.

There's nothing stopping you from initializing the Formie JS on your own though. You can render a form without auto-initializing the JS, and then initialize it yourself depending on your logic. However, the're no split between Formie's JS for non-tracking things and tracking-things like captchas, so that may be problematic.

**Reply by thisisjamessmith** [reply]
Ah damn ok. I'll see if we can sell in Friendly Captcha which apparently doesn't require consent.
